src/win/pipe.c

```diff
@@ -753,9 +764,14 @@ int uv_pipe_bind2(uv_pipe_t* handle,
     req->next_pending = NULL;
   }
 
-  err = uv__convert_utf8_to_utf16(name, &handle->name);
-  if (err)
-    return err;
+  /* TODO(bnoordhuis) Add converters that take a |length| parameter. */
```

Comment:
Instead of leaving this as a TODO, how about we replace the call `if (*name == '\0') {` above with `if (memchr(name, '\0', namelen) != NULL)` instead (and add an equivalent call to unix, but only if `name[0] != '\0'`)

Reply:
For my understanding, you're worried about truncation attacks?

The TODO is about uv__convert_utf8_to_utf16() expecting nul terminated input, which necessitates the malloc+copy.

Reply:
yeah, I was thinking of truncation attacks, and observing that it wouldn't matter if it was null terminated. The buffer is only about 256 bytes max, so it doesn't really need the malloc, just a copy to a stack buffer

Reply:
Updated. I hope I understood you correctly.